shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -443,6 +439,32 @@ export class CoreUtils {
 
         return result;
     }
+
+    /**
+     * Return the current time via the Date now() function (if available) and falls back to (new Date()).getTime() if now() is unavailable (IE8 or less)
+     * https://caniuse.com/#search=Date.now
+     */
+    public static dateNow() {
+        let dt = Date;
+        if (dt.now) {
+            return dt.now();
+        }
+
+        return new dt().getTime();
+    }
+
+    /**
+     * Return the current value of the Performance Api now() function (if available) and fallback to CoreUtils.dateNow() if it is unavailable (IE9 or less)
+     * https://caniuse.com/#search=performance.now
+     */
+    public static perfNow() {
+        let perf = getPerformance();
+        if (perf && perf.now) {
+            return perf.now();
+        }
+
+        return CoreUtils.dateNow();
```

Comment:
Maybe this is something we discussed before, but date and perf represent time differently so returning CoreUtils.dateNow here could cause conflicts with the data, maybe returning 0 because is not supported

Reply:
This is based on previous fallback scenarios, so when the performance api is not available it falls back to providing something even at the lower resolution